pgraph/pgraph_test.go

```diff
@@ -840,3 +840,49 @@ func TestDeleteEdge2(t *testing.T) {
 		t.Errorf("expected outgoing graph edges for vertex v1: e4, instead of: %s", res)
 	}
 }
+
+func TestFindEdge(t *testing.T) {
+	g, _ := NewGraph("g")
+
+	v1 := NV("v1")
+	v2 := NV("v2")
+	v3 := NV("v3")
+	v4 := NV("v4")
+	v5 := NV("v5")
+	v6 := NV("v6")
+	v7 := NV("v7")
+
+	e1 := NE("e1")
+	e2 := NE("e2")
+	e3 := NE("e3")
+	e4 := NE("e4")
+	e5 := NE("e5")
+	e6 := NE("e6")
+	e7 := NE("e7")
+	e8 := NE("e8")
+	e9 := NE("e9")
+	e10 := NE("e10")
+
+	g.AddEdge(v1, v2, e1)
+	g.AddEdge(v1, v4, e2)
+	g.AddEdge(v1, v3, e3)
+	g.AddEdge(v2, v3, e4)
+	g.AddEdge(v2, v4, e5)
+	g.AddEdge(v2, v6, e6)
+	g.AddEdge(v2, v5, e7)
+	g.AddEdge(v3, v6, e8)
+	g.AddEdge(v4, v7, e9)
+	g.AddEdge(v5, v7, e10)
+
+	if !(g.HasVertex(v1) && g.HasVertex(v4) && g.HasVertex(v7)) {
+		t.Errorf("graph expected to have vertices v1, v4, and v7")
+		return
```

Comment:
We could return here if it would affect future tests, but it won't. So either add a return for the other two cases, or remove this one. I think the latter is best =D

Reply:
If this test fails, that means that v1, v4, or v7 doesn't exist, in which case the condition of one of the following if statements will create an error. In that case we don't want to continue to the other cases. Let me know what you think.

Reply:
Agreed, however it's not necessarily harmful. It's fine to just "return" and stop the test early. But be consistent.

Reply:
Okay, I'll follow your suggestion in that case!